From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Li Hua Qian <huaqian.li@siemens.com>
Date: Thu, 11 Sep 2025 19:37:06 +0800
Subject: [PATCH] Fix the prototypes building issue

The kernel has stricter enforcement of function prototypes starting
around v6.2, Linux 6.12 enables -Wmissing-prototypes by default in the
kernel build system (via KBUILD_CFLAGS). This flag requires explicit
function declarations for all non-static functions.

Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
---
 common/pcie_common.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/pcie_common.h b/common/pcie_common.h
index 642391c58999..0c2017e3475f 100644
--- a/common/pcie_common.h
+++ b/common/pcie_common.h
@@ -134,6 +134,7 @@ int hailo_pcie_write_firmware(struct hailo_pcie_resources *resources, const void
 int hailo_pcie_write_firmware_batch(struct device *dev, struct hailo_pcie_resources *resources, u32 stage);
 bool hailo_pcie_is_firmware_loaded(struct hailo_pcie_resources *resources);
 bool hailo_pcie_wait_for_firmware(struct hailo_pcie_resources *resources);
+bool hailo_pcie_is_device_ready_for_boot(struct hailo_pcie_resources *resources);
 
 int hailo_pcie_read_firmware_notification(struct hailo_pcie_resources *resources,
     struct hailo_d2h_notification *notification);
@@ -165,4 +166,4 @@ void hailo_soc_write_soc_connect(struct hailo_pcie_resources *resources);
 }
 #endif
 
-#endif /* _HAILO_COMMON_PCIE_COMMON_H_ */
\ No newline at end of file
+#endif /* _HAILO_COMMON_PCIE_COMMON_H_ */
